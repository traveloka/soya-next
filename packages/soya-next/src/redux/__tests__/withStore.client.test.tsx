import React from "react";
import { render } from "@testing-library/react";
import setupTest from "./utils/withStoreSetup";

describe("withStore - browser", () => {
  it("should add store to page props", async () => {
    const { PageWithStore } = setupTest();
    const props = await PageWithStore.getInitialProps({} as any);
    const { getByTestId } = render(
      <PageWithStore {...props} store={{ soya: true } as any} />
    );
    const propsStore = getByTestId("props-store");
    const propsInit = getByTestId("props-init");
    expect(propsStore).toBeInTheDocument();
    expect(propsInit).toBeInTheDocument();
    expect(propsStore).toMatchSnapshot();
    expect(propsInit).toMatchSnapshot();
  });
});
