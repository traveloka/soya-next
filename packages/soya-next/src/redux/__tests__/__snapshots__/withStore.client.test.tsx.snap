// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withStore - browser should add store to page props 1`] = `
<div
  data-testid="props-store"
>
  {
  "soya": true
}
</div>
`;

exports[`withStore - browser should add store to page props 2`] = `
<div
  data-testid="props-init"
>
  true
</div>
`;
